src/main/java/io/orangebeard/listener/OrangebeardTestSystemListener.java

```diff
@@ -45,107 +45,144 @@
 import static java.util.Objects.requireNonNull;
 
-@NoArgsConstructor
 public class OrangebeardTestSystemListener implements TestSystemListener, Closeable {
-    private String propertyFileName = "orangebeard.properties";
+    private final Logger logger = LoggerFactory.getLogger(OrangebeardTestSystemListener.class);
+
     private static final String PROP_ROOT_PATH = "fitnesseroot.path";
     private static final String PROP_ATTACH_ZIP = "attach.zipfile";
+    private final boolean attachZip;
+    private final String rootPath;
+    private String propertyFileName = "orangebeard.properties";
+    private boolean local = false;
+    private static int numberOfLogs = 0;
 
-    private final Logger logger = LoggerFactory.getLogger(OrangebeardTestSystemListener.class);
-
+    private final OrangebeardProperties orangebeardProperties;
+    private final ScenarioLibraries scenarioLibraries;
     private OrangebeardClient orangebeardClient;
-    private final OrangebeardProperties orangebeardProperties = new OrangebeardProperties();
     private ToolchainRunningContext runContext;
+    private AttachmentHandler attachmentHandler;
 
-    private String rootPath = getFitnesseRootPath();
-    private boolean attachZip = false;
-    private boolean local = false;
-    private OrangebeardLogger orangebeardLogger;
+    /**
+     * constructor for testing purposes
+     */
+    OrangebeardTestSystemListener(
+            OrangebeardProperties orangebeardProperties,
+            ToolchainRunningContext runContext,
+            OrangebeardClient orangebeardClient,
+            AttachmentHandler attachmentHandler,
+            ScenarioLibraries scenarioLibraries) {
+        this.orangebeardProperties = orangebeardProperties;
+        this.runContext = runContext;
+        this.orangebeardClient = orangebeardClient;
+        this.attachmentHandler = attachmentHandler;
+        this.scenarioLibraries = scenarioLibraries;
+        this.attachZip = false;
+        this.rootPath = getFitnesseRootPath();
+    }
+
+    public OrangebeardTestSystemListener() {
+        this.orangebeardProperties = new OrangebeardProperties();
+        this.scenarioLibraries = new ScenarioLibraries();
+        this.attachZip = false;
+        this.rootPath = getFitnesseRootPath();
+    }
 
     public OrangebeardTestSystemListener(@Nullable String propertyFileName, String rootPath) {
         if (propertyFileName != null) {
             this.propertyFileName = propertyFileName;
         }
-        determineResultZip();
+        this.attachZip = attachZip();
         this.rootPath = rootPath;
+        this.orangebeardProperties = new OrangebeardProperties();
+        this.scenarioLibraries = new ScenarioLibraries();
     }
 
     public OrangebeardTestSystemListener(String propertyFileName, boolean local) {
         this.propertyFileName = propertyFileName;
-        determineResultZip();
+        this.attachZip = attachZip();
         this.local = !local;
```

Comment:
@tcnh You added this as far as I can see. Can you remember in any way what this property does, and why it is so confusing here? Naming should be better...